rockar06/HeroesApp
# #39 | Create rules module to create custom lint rules for project

### Summary

This PR adds support for custom lint rules, in this case, it is added the rule: `ExcludedClassIfHiltModuleRule` that only allows the usage of `ExcludeClassFromJacocoGeneratedReport` annotation in abstract classes used as Hilt Modules.  

### Steps to verify changes


1. Create a test class and annotate with ` ExcludeClassFromJacocoGeneratedReport`
2. Run `./gradlew build` command
3. Verify there is a lint error while building project
4. Remove ` ExcludeClassFromJacocoGeneratedReport` annotation
5. Run `./gradlew build` command
6. Project should build successfully 


**Thanks for your time!**
